### DISCUSSION TITLE
🚀 FR Add a data integrity check to all form submissions

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3750] [Feature Requests]
### Suggest an idea for ExpressionEngine

This comes up often enough to be a problem.  With complex entry forms, it's not hard to hit the default max_input_vars setting of 1000.  Fluid/grid fields with a decent number of columns can really pile on the form fields.  And when you hit the max, it just drops the extra variables.

This can result in data loss if you save something and the last of the data is simply dropped.  Or more often, it results in just really whacky behavior.

I just debugged one where when you changed a radio select it threw a validation error on the field AND actually saved the entry.  (When you change a radio, it triggers a post.) 

You couldn't easily tell it was saving the entry, but that's what it was doing, and the unexpected banner data coming back weirdly caused a validation error to appear.  Bumped the max_vars up, problem solved.  After WAY too much time going down rabbit holes.

If we could automatically add a last field to all forms or a count of submitted fields- something of that type- and check for it and return an error message instead of treating it as a legit form submission, that would save a lot of trouble.

**Comment by intoeetive** [comment]
IDK if injecting check into every form submission is worth doing... Perhaps PHP health check utility would work better? Like it checks the number of your custom fields for each channel and if it's more than 80% of your max_input_vars it shows a warning... and maybe few other checks?
